db/migrate/20211222192244_update_combined_advanced_filters.rb

```diff
@@ -0,0 +1,188 @@
+      type: 'search'
+    },
+    'address-usa': {
+      name: 'address-usa',
+      title: 'Address (within USA) (Text)',
+      description: 'Monitoree Address 1, Town/City, State, Address 2, Zip, or County within USA',
+      type: 'search'
+    },
+    'first-name': {
+      name: 'first-name',
+      title: 'Name (First) (Text)',
+      description: 'Monitoree first name',
+      type: 'search'
+    },
+    'last-name': {
+      name: 'last-name',
+      title: 'Name (Last) (Text)',
+      description: 'Monitoree last name',
+      type: 'search'
+    },
+    'middle-name': {
+      name: 'middle-name',
+      title: 'Name (Middle) (Text)',
+      description: 'Monitoree middle name',
+      type: 'search'
+    },
+    'telephone-number': {
+      name: 'telephone-number',
+      title: 'Primary Contact Telephone Number (Exact Match) (Text)',
+      description: 'Monitorees with a specified 10 digit primary contact telephone number',
+      type: 'search'
+    },
+    'telephone-number-partial': {
+      name: 'telephone-number-partial',
+      title: 'Primary Contact Telephone Number (Contains) (Text)',
+      description: 'Monitorees with a primary contact telephone number that contains specified digits',
+      type: 'search'
+    }
+  }
+
+  def up
+    ActiveRecord::Base.transaction do
+      # Migrate saved advanced filters
+      relevant_filters.each do |uf|
+        contents = JSON.parse(uf[:contents])
+        migrate_advanced_filter_contents(contents)
+        uf.update!(contents: contents.to_json)
+      end
+
+      # Migrate advanced filters in saved export presets
+      relevant_export_presets.each do |uep|
+        config = JSON.parse(uep[:config])
+        contents = config.dig('data', 'patients', 'query', 'filter')
+        migrate_advanced_filter_contents(contents)
+        config['data']['patients']['query']['filter'] = contents
+        uep.update!(config: config.to_json)
+      end
+    end
+  end
+
+  def down
+    ActiveRecord::Base.transaction do
+      # Rollback saved advanced filters
+      relevant_filters.each do |uf|
+        contents = JSON.parse(uf[:contents])
+        rollback_advanced_filter_contents(contents)
+        uf.update!(contents: contents.to_json)
+      end
+
+      # Rollback advanced filters in saved export presets
+      relevant_export_presets.each do |uep|
+        config = JSON.parse(uep[:config])
+        contents = config.dig('data', 'patients', 'query', 'filter')
+        rollback_advanced_filter_contents(contents)
+        config['data']['patients']['query']['filter'] = contents
+        uep.update!(config: config.to_json)
+      end
+    end
+  end
+
+  def migrate_advanced_filter_contents(contents)
+    contents.each do |content|
+      filter_name = content['filterOption']['name'].to_sym
+      if %i[address-foreign address-usa first-name last-name middle-name].include?(filter_name)
```

Comment:
For the address filters, I'm seeing them not combine into one filter and the values not persisted:

<img width="1166" alt="Screen Shot 2022-02-04 at 6 13 51 PM" src="https://user-images.githubusercontent.com/35042815/152615493-92ce56b7-26c0-4b05-af8e-fb01230f0e3e.png">

Reply:
Same with the names...

<img width="1166" alt="Screen Shot 2022-02-04 at 6 13 51 PM" src="https://user-images.githubusercontent.com/35042815/152615610-fe989603-9694-4242-8932-e556768e204c.png">

Reply:
I guess this is a question for @hackrm... should these be made into a combination filter together or be kept separate on the migration?  I would think the former would make sense but would get complicated if there were two first names in there

Reply:
I intentionally kept these separate because I think it might cause confusion if we combine them. If we leave them as is, the behavior of the saved filters will remain the same as the users expect whereas if we combine them, users might be seeing filters that they didn't want combined (ex: user has one filter should search for first name 'john' and a different filter with last name 'doe', the user shouldn't suddenly just have those two filters applied at the same time, this is even more so the case for address inside/outside USA)

Reply:
Yeah... I could make an argument for either option but we can leave as is for simplicity